vaadin-testbench-integration-tests-junit5/src/test/java/com/vaadin/tests/ElementQueryIT.java

```diff
@@ -305,4 +307,163 @@ void allWithoutLightTheme() {
         Assertions.assertEquals(4, allButtons.size());
     }
 
+    @BrowserTest
+    public void labelMatches() {
+        openTestURL();
+        TemplateViewElement view = $(TemplateViewElement.class).waitForFirst();
+
+        List<LabelPlaceholderElement> labelPlaceholderElements = view.$(LabelPlaceholderElement.class)
+                .withLabel("one")
+                .all();
+        Assertions.assertEquals(2, labelPlaceholderElements.size());
+
+        labelPlaceholderElements = view.$(LabelPlaceholderElement.class)
+                .withLabel("two")
+                .all();
+        Assertions.assertEquals(2, labelPlaceholderElements.size());
```

Comment:
I would probably change the view so that search for `one` and `two` returns a different number of items. Having the same assertion may hide a bug.
Also, it could be good to add a search for "none", asserting it gets an empty collection.